cohortextractor/query_engines/mssql.py

```diff
@@ -378,6 +390,31 @@ def get_case_expression(self, mapping, default):
             else_=default,
         )
 
+    def get_expression_for_value_from_function(self, value):
+        argument_expressions = []
+        tables = set()
+        for arg in value.arguments:
+            arg_expr, arg_tables = self.get_value_expression(arg)
+            argument_expressions.append(arg_expr)
+            tables.update(arg_tables)
+        method_name = f"apply_function_{value.function}"
+        method = getattr(self, method_name, None)
+        if not method:
+            raise ValueError(f"Unsupported function: {value.function}")
+        value_expression = method(*argument_expressions)
+        return value_expression, tuple(tables)
+
+    def apply_function_date_diff_in_years(self, start_date, end_date):
+        # `literal_column` doesn't seem quite the right construct here, but I
+        # need SQLAlchemy to generate the string "year" without quotes, and
+        # this seems to do the trick
+        YEAR = sqlalchemy.literal_column("year")
+        year_diff = sqlalchemy.func.datediff(YEAR, start_date, end_date)
```

Comment:
Does datediff round up to the nearest int?

Reply:
It diffs the year part but ignores month and day entirely

Reply:
It diffs the year part but ignores month and day entirely

Reply:
Ah OK.  I see the reason for the -1 calculation then :+1: